app/build_svc.py

```diff
@@ -191,10 +207,13 @@ def _replace_args_props(self, ability):
         :rtype: string
         """
         env = self.get_config(prop='enabled', name='build').get(ability.language)
-        cmd = env['entrypoint_args'].replace('#{code}', self.build_file)
-        cmd = cmd.replace('#{build_target}', ability.build_target)
+        build_file = '{}.{}'.format(self.build_file, env['extension']) if env.get('extension') else build_file
+        build_command = env['build_command'].replace('#{code}', build_file)
+        build_command = build_command.replace('#{build_target}', ability.build_target)
+
+        if ability.language == 'csharp':
+            references = [p for p in ability.payloads if p.endswith('.dll')]
```

Comment:
move to its own function

Reply:
Moved references replacement to separate function (hoping this is what you meant)